sync/index.test.ts

```diff
@@ -226,6 +227,79 @@ test('maps input actions', async () => {
   equal(pair.rightNode.log.actions(), [{ type: 'a1' }])
 })
 
+test('handles error in onSync', async () => {
+  let error = new Error('test')
+  let catched: Error[] = []
+
+  let pair = await createTest()
+  pair.rightNode.options.onSync = () => {
+    throw error
+  }
+  pair.rightNode.catch(e => {
+    catched.push(e)
+  })
+  pair.leftNode.log.add({ type: 'a' })
+
+  await delay(50)
+  equal(catched, [error])
+})
+
+test('onSync is called instead of `inMap`, `inFilter` and `Log#add`', async () => {
+  let actions: Action[] = []
+  let pair = await createTest(created => {
+    created.rightNode.options.inFilter = async action => {
+      return action.type !== 'c1'
+    }
+    created.rightNode.options.inMap = async (action, meta) => {
+      return [{ type: action.type + '1' }, meta]
+    }
+    created.rightNode.options.onSync = (_, action) => {
+      actions.push(action)
```

Comment:
You need to change test to be sure that `processAction` works.

You can put action to the array and then call adding in a different order (or do some other changes).